geiger/geiger.py

```diff
@@ -17,41 +17,53 @@
 import os
 os.environ['OMP_NUM_THREADS'] = '4'
 
-
-EMBEDDING_FILE = '../input/fasttext-crawl-300d-2m/crawl-300d-2M.vec'
-
-train = pd.read_csv('../input/jigsaw-toxic-comment-classification-challenge/train.csv')
-test = pd.read_csv('../input/jigsaw-toxic-comment-classification-challenge/test.csv')
-submission = pd.read_csv('../input/jigsaw-toxic-comment-classification-challenge/sample_submission.csv')
-
-X_train = train["comment_text"].fillna("fillna").values
-y_train = train[["toxic", "severe_toxic", "obscene", "threat", "insult", "identity_hate"]].values
-X_test = test["comment_text"].fillna("fillna").values
-
-
-max_features = 30000
-maxlen = 100
-embed_size = 300
-
-tokenizer = text.Tokenizer(num_words=max_features)
-tokenizer.fit_on_texts(list(X_train) + list(X_test))
-X_train = tokenizer.texts_to_sequences(X_train)
-X_test = tokenizer.texts_to_sequences(X_test)
-x_train = sequence.pad_sequences(X_train, maxlen=maxlen)
-x_test = sequence.pad_sequences(X_test, maxlen=maxlen)
-
-
-def get_coefs(word, *arr): return word, np.asarray(arr, dtype='float32')
-embeddings_index = dict(get_coefs(*o.rstrip().rsplit(' ')) for o in open(EMBEDDING_FILE))
-
-word_index = tokenizer.word_index
-nb_words = min(max_features, len(word_index))
-embedding_matrix = np.zeros((nb_words, embed_size))
-for word, i in word_index.items():
-    if i >= max_features: continue
-    embedding_vector = embeddings_index.get(word)
-    if embedding_vector is not None: embedding_matrix[i] = embedding_vector
-
+import argparse
+
+parser = argparse.ArgumentParser(description='The geiger programm')
+parser.add_argument('train', type=str, help='The csv with the input data')
+parser.add_argument('test', type=str, help='The csv with the test data')
+parser.add_argument('embeddings', type=str, help='The embeddings file (e.g. fasttext)')
+parser.add_argument('--features', type=int, default=30000, help='The numbers of features of the embedding file. (in the first line of the file)')
+parser.add_argument('--maxlen', type=int, default=100, help='The length of the max sequence')
+parser.add_argument('--embed_size', type=int, default=300, help='The size of the embedding layer')
+parser.add_argument('--batch_size', type=int, default=32, help='Batch size ')
+parser.add_argument('--epochs', type=int, default=2, help='Total number of epochs to run for')
+
+args = parser.parse_args()
+
+
+# EMBEDDING_FILE = '../input/fasttext-crawl-300d-2m/crawl-300d-2M.vec'
+
+def load_files():
```

Comment:
maybe this can be split up into two functions: load_kagle_toxic datasets, which uses pandas to load the datasets and return the X_train, Y_train and X_test after calling `.values` on them.
Then another function that does the feature extraction. In this case we are using the tokenizer to get some (padded) word sequences